### PR TITLE
Make compatible with Python-3

### PR DESCRIPTION
**Pull request by wosc**
Note that you'll have to escape `git(hub)-notifier.cfg` settings like `(notifier-)emailprefix = [%r]` as `%%r` under Python-3, because ConfigParser now attempts some kind of expansion otherwise.

### CODE REVIEW

**Comment by rsmmr** [comment]
Thanks, looks good! The ConfigParser incompatibility concerns me a bit though, as that means a config file will work correctly only with one of the two versions, which seems could be confusing (and problematic for existing config files). Any idea if there's a way to get around that problem and make the old syntax work with Python 3?

**Comment by wosc** [comment]
As it turns out, this interpolation feature can indeed be disabled. Now it works just like it always did.

**Comment by rsmmr** [comment]
Great, thanks.

 I was just about to merge it, however I ran into a couple more problems. One is just a typo in a format string, I'll add a comment for that. The 2nd may be a larger problem though: In Python 2, the `write` calls don't seem to seem to deal with the non-ASCII characters the same way as the `print` statements used to do. Example: A commit with this author: ``Author: Röbin Såmmer <robin@icir.org>`` triggers this: 

```

Traceback (most recent call last):
  File "../git-notifier", line 844, in <module>
    reportPath(current, new_revs)
  File "../git-notifier", line 749, in reportPath
    commit(current, rev, force=force, subject_head=subject_head)
  File "../git-notifier", line 703, in commit
    sendChangeMail(rev, subject, heads, show_cmd, diff_cmd, stat_cmd)
  File "../git-notifier", line 649, in sendChangeMail
    out.write(line + "\n")
UnicodeEncodeError: 'ascii' codec can't encode character u'\xf6' in position 9: ordinal not in range(128)
```

This specific instance would be easy to fix, but that looks like it could be a more general problems at other places, too. Any idea?

File: git-notifier [hunk 1]
```diff
@@ -296,15 +295,15 @@ def git(args, stdout_to=subprocess.PIPE, all=False):
 
     try:
         if Config.debug:
-            print >>sys.stderr, "> git " + args
```
Comment (rsmmr) [inline git-notifier:299]: `%s`

**Comment by wosc** [comment]
Yeah, I'm somewhat familiar with the unicode handling differences between 2 and 3 (and how to mitigate them), but they are kind of tricky to spot by just reading through the code (also, some APIs changed predictably, some... didn't)  -- do you have some kind of test setup or a testing repository on which you run git-notifier to check whether stuff works?

**Comment by rsmmr** [comment]
Nothing systematic unfortunately. I have a local git repository with a bunch of dummy commits that I use to test changes with, but it's all manual and adhoc. If you want I can send you a copy of that, is the email address from your profile the right contact?

**Comment by wosc** [comment]
Yes, that would be excellent, thank you!

**Comment by rsmmr** [comment]
Sent by email.

**Comment by wosc** [comment]
Alright. Sorry for the delay, I was travelling. I can reproduce the UnicodeEncodeError you mentioned with your test repository easily under Python-2 (Python-3 is fine, ironically enough) -- I think that in my own test runs I simply didn't have any non-ascii characters anywhere (author names, diffs, etc.), which to be fair was somewhat of an oversight. ;)

The reason for the error is the typical 2->3 change that many APIs (in this case `subprocess.Popen`) under Python-3 work with unicode objects, while under Python-2 they work with bytes objects. I remember that Python-3 complained immediately, and thus I added a `decode('utf-8')` in the `git()` function, but did not realize that this would break under Python-2 in non-ascii cases. The proper fix then is to only decode to unicode under py-3, and leave py-2 well enough alone.

I've now run the test repository forwards and backwards (e.g. `--manual 1ae6db4ab2fc8b9b12dc0cbc485e6dd3aa4d6396..HEAD --maxage 0 --debug`), with both py-3 and py-2, and everything seems fine to me. :)

File: git-notifier (hunk 1 above)
Comment (wosc) [reply git-notifier:299]: Well spotted, I've fixed this via amend/rebase.

**Pull request merged**